subsys/bluetooth/common/log.h

```diff
@@ -41,7 +41,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #define BT_INFO(fmt, ...) LOG_INF(fmt, ##__VA_ARGS__)
 
 #define BT_ASSERT(cond) if (!(cond)) { \
-				BT_ERR("assert: '" #cond "' failed"); \
+				printk("assert: '" #cond "' failed"); \
```

Comment:
Why not calling __ASSERT instead of printk ? Then we have a common handling of asserts.

Reply:
Because `__ASSERT()` is disabled by default, and here we want it enabled by default, since it is critical for us to find issues in the wild.

Reply:
OK :-)
I do not like having printk called directly without having a way of compiling with asserts without spending the memory on the printk strings. Could it somehow be possible to make it configurable if the printk part of BT_ASSERT should be enabled.

Reply:
Yes, actually I was going to add this as a Kconfig option since one can usually figure out the guilty line by using a debugger in `k_panic()`. I will send a proposal to this PR

Reply:
:-)